frontend/src/components/layouts/AppLayout.tsx

```diff
@@ -91,6 +91,10 @@ export default function AppLayout() {
     return null;
   }
 
+  if (!info.setupCompleted || !info.running || !info.unlocked) {
```

Comment:
This looks unsafe to me. What about we move this info check
```
if (!info) {
  return null;
}
```

Into App.tsx e.g.
`{info && <RouterProvider router={router} />}`

Reply:
👍 yup that would be good, but line 94 does the fix here, should we also move it to App.tsx is what you mean?

Reply:
Line 94 with any of these conditions not being true feels worrying to me - it could break one of our pages

Reply:
But if any of it is false, we do the redirect in `HomeRedirect`. `Outlet` is anyways rendered even now. But it is just rendered with the Sidebar. (And it consists of the `HomeRedirect` component which then transfers to the intro/start/unlock pages) This change just removes the Sidebar in those cases by rendering just the `Outlet` (`HomeRedirect` component)

Reply:
@im-adithya this makes it more complicated and does not cleanly solve the problem. We don't even know what page to render if the info is not loaded. Rendering the outlet makes no sense